src/Avalonia.Base/StyledElement.cs

```diff
@@ -590,6 +637,30 @@ protected virtual void OnInitialized()
         {
         }
 
+        protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
+        {
+            base.OnPropertyChanged(change);
+
+            if (change.Property == ThemeProperty)
+            {
+                // Changing the theme detaches all styles, meaning that if the theme property was
+                // set via a style, it will get cleared! To work around this, if the value was
+                // applied at less than local value priority then promote the value to local value
+                // priority until styling is re-applied.
+                if (change.Priority > BindingPriority.LocalValue)
```

Comment:
comment says less than, but condition is greater than... just checking my understanding here assume its probably correct.

Reply:
This enum goes in opposite way. Lower priority values have higher integer value.